stage #1/module #3. Generics/lesson #1. Overview/content.md

```diff
@@ -0,0 +1,71 @@
+# Generics
+
+## Why Use Generics?
+In a nutshell, generics enable types (classes and interfaces) to be parameters when defining classes, interfaces and
+methods. Much like the more familiar formal parameters used in method declarations, type parameters provide a way for
+you to re-use the same code with different inputs. The difference is that the inputs to formal parameters are values,
+while the inputs to type parameters are types.
+
+Code that uses generics has many benefits over non-generic code:
+
+- Stronger type checks at compile time.
+- Elimination of casts.
+- Enabling programmers to implement generic algorithms.
+
+## Type Parameters in Java Generics
+The type parameters naming conventions are important to learn generics thoroughly. The common type parameters are as
+follows:
+
+- T – Type
+- E – Element
+- K – Key
+- N – Number
+- V – Value
+
+## Type Erasure
+Generics were introduced to the Java language to provide tighter type checks at compile time and to support generic
+programming. To implement generics, the Java compiler applies type erasure to:
```

Comment:
I am suggesting to markup **type erasure**, as it it very important phrase to remember